File: 16/03/tests/login_test.py
import pytest
from pages import login_page


class TestLogin():

    @pytest.fixture
    def login(self, driver):
        return login_page.LoginPage(driver)

    @pytest.mark.shallow
    def test_valid_credentials(self, login):
        login.with_("tomsmith", "SuperSecretPassword!")
        assert(login.success_message_present())

    def test_invalid_credentials(self, login):
        login.with_("tomsmith", "bad password")
        assert(login.failure_message_present())
        # assert(self.login.success_message_present() == False)

    @pytest.mark.shallow
    def test_forced_failure(self, login):
        login.with_("tomsmith", "bad password")
        assert(login.success_message_present())

File: javascript/16/03/test/LoginTest.js
require('./spec_helper')
const assert = require('assert')
const LoginPage = require('../pages/LoginPage')

describe('Login', function() {
  let login

  beforeEach(async function() {
    login = new LoginPage(this.driver)
    await login.load()
  })

  it('with valid credentials @shallow', async function() {
    await login.authenticate('tomsmith', 'SuperSecretPassword!')
    assert(await login.successMessagePresent(), 'Success message not displayed')
  })

  it('with invalid credentials @deep', async function() {
    await login.authenticate('tomsmith', 'bad password')
    assert(await login.failureMessagePresent(), 'Failure message not displayed')
  })

  it('forced failure @shallow', async function() {
    await login.authenticate('tomsmith', 'bad password')
    assert.equal(false, true)
  })
})
